### PR TITLE
Remove dead-code related to yield tests

### PR DESCRIPTION
**Pull request by nicoddemus**
Just noticed some code that no longer is needed when we removed yield-tests

### CODE REVIEW

**Comment by nicoddemus** [comment]
*sigh* no idea why codecov/patch claims it has only reached 71.42% of the target diff, while codecov/project says it has increased coverage by 0.05%.

**Comment by blueyed** [comment]
> no idea why codecov/patch claims it has only reached 71.42% of the target diff, while codecov/project says it has increased coverage by 0.05%.

That does not contradict itself.

Regarding the timeout on @codecov it seems like we do not have https://github.com/pytest-dev/pytest/pull/4799 in features yet?

**Comment by blueyed** [comment]
> Regarding the timeout on @codecov it seems like we do not have #4799 in features yet?

Hmm, does not look like it?!  https://github.com/pytest-dev/pytest/pull/4833

File: src/_pytest/python.py [hunk 1]
```diff
@@ -1419,16 +1414,11 @@ def __init__(
...
-            if hasattr(self, "callspec"):
-                callspec = self.callspec
-                assert not callspec.funcargs
-                if hasattr(callspec, "param"):
-                    self.param = callspec.param
+        if hasattr(self, "callspec"):
+            callspec = self.callspec
+            assert not callspec.funcargs
+            if hasattr(callspec, "param"):
+                self.param = callspec.param
```
Comment (blueyed) [inline src/_pytest/python.py]: This line appears to not be covered.. I've added `assert 0, 2` before it.
It was not covered previously then also.  https://codecov.io/gh/pytest-dev/pytest/src/3fd2f43fb6b7bcd2eb342d0d7056381937d3d8db/src/_pytest/python.py#L1431

File: src/_pytest/debugging.py [hunk 2]
```diff
@@ -185,17 +185,12 @@ def _test_pytest_function(pyfuncitem):
...
-        pyfuncitem._args = tuple(arg_list)
-    else:
-        if "func" in pyfuncitem._fixtureinfo.argnames:
-            raise ValueError("--trace can't be used with a fixture named func!")
-        pyfuncitem.funcargs["func"] = testfunction
-        new_list = list(pyfuncitem._fixtureinfo.argnames)
-        new_list.append("func")
-        pyfuncitem._fixtureinfo.argnames = tuple(new_list)
+    if "func" in pyfuncitem._fixtureinfo.argnames:
+        raise ValueError("--trace can't be used with a fixture named func!")
```
Comment (blueyed) [inline src/_pytest/debugging.py]: Not covered.  Also previously: https://codecov.io/gh/pytest-dev/pytest/src/3fd2f43fb6b7bcd2eb342d0d7056381937d3d8db/src/_pytest/debugging.py

**Comment by nicoddemus** [comment]
> That does not contradict itself.

I just read https://docs.codecov.io/docs/commit-status#section-patch-status now and I think I understand that better now, shame on me for not looking at it before. But I still find it strange in this case: I removed some code, including "if" statements whose body were not executing at all previously. I also see your comments on the code and they were not being covered before anyway, I don't see why my patch coverage could have been reduced.

[Other people also seem to be confused by this](https://github.com/kata-containers/tests/issues/508).
Perhaps we should consider disabling the `codecov/patch` check and only leave `codecov/project` (https://github.com/kata-containers/runtime/pull/512)? Then we can make a stance to never let coverage decrease on PRs. As it stands, I see `codecov/patch` fail with some frequency, and we always ignore it anyways. What do you think @blueyed? 

> Regarding the timeout on @codecov it seems like we do not have #4799 in features yet?

Indeed we do not:

```
λ git branch -a --contains 1ec7f60484e85f96b994dba4ba4825227f6a17c1
  remotes/origin/codocov-azure
  remotes/upstream/master
```

Which is a good sign. 👍 

Thanks for the explanations @blueyed, appreciate it!

**Comment by blueyed** [comment]
> I don't see why my patch coverage could have been reduced.

It does not say it reduced, but just how much it is covered.
If you just added whitespace around the previously uncovered code it would be the same.

What this tries to enforce is that a patch/diff is at least covered as much as the project.
I usually require patches to be covered 100% (regardless of the project's coverage percentage) - you are touching some code, and usually it makes sense to ensure that it is tested.. :)

**Comment by blueyed** [comment]
> As it stands, I see `codecov/patch` fail with some frequency, and we always ignore it anyways.

Isn't that about codecov/project (flaky coverage in general)?

**Comment by nicoddemus** [comment]
> It does not say it reduced, but just how much it is covered.
If you just added whitespace around the previously uncovered code it would be the same.

I see, thanks.

I will rebase this after https://github.com/pytest-dev/pytest/pull/4833 gets merged, I want to understand how I can fix the patch coverage.

**Comment by nicoddemus** [comment]
> Isn't that about codecov/project (flaky coverage in general)?

I'm under the impression that's usually `codecov/patch` that does this, but I'm not sure anymore.

**Comment by blueyed** [comment]
You can rebase it now.

**Comment by nicoddemus** [comment]
Done, thanks for the ping. 👍

**Comment by nicoddemus** [comment]
Hmmm I still can't access the `codecov/patch` link: https://codecov.io/gh/pytest-dev/pytest/compare/c7bbb2a78873a63b9df9df08f8dc77cde30debd3...a3334783cb9432fabb87ff227c79981e41e661dc

I can't access `codecov/project` either for that matter. Should we wait a little bit for `codecov` to finish processing the data?

**Comment by blueyed** [comment]
> Should we wait a little bit for `codecov` to finish processing the data?

You can try it several times, it might work then.
But in general it will likely only show what I've told you before (plus maybe some more).

**Comment by nicoddemus** [comment]
> But in general it will likely only show what I've told you before (plus maybe some more).

If the link it posts does not lead to an easy to use diff, I'm afraid it is not really useful in the end right?

What can we do to improve this situation? 

Perhaps post just a few select environments, for example `py27` and `py37`? It will show false negatives (for example tests which depend on `nympy` won't run and will be reported as not covered), but it is better to have some diff available for the usual cases.

Try another service, like Code Climate or Coveralls?

Sorry if I seem bitter, I definitely acknowledge the importance of having code coverage and really appreciate your efforts on that front, I'm just a little annoyed that we are not getting useful information out of this.

**Comment by blueyed** [comment]
> What can we do to improve this situation?

1. Remove coverage for AppVeyor.
2. Remove unnecessary/duplicate reporting.

Started on this in https://github.com/pytest-dev/pytest/pull/4839.

**Comment by blueyed** [comment]
> Sorry if I seem bitter, I definitely acknowledge the importance of having code coverage and really appreciate your efforts on that front, I'm just a little annoyed that we are not getting useful information out of this.

I also hate this, and that codecov is taking since September last year to handle this already, although they should clearly have this in their logs etc.

(On the positive side the numbers at least appear to be correct, i.e. it can tell you if you miss something or not, and you can then check the diff from GitHub and their browsing based on the commit.)

**Comment by nicoddemus** [comment]
> 1. Remove coverage for AppVeyor.

Oh right, that's a no brainer. 👍 

> 2. Remove unnecessary/duplicate reporting.

Indeed. I will see how to make this easier on Azure, let's see how #4839 fares then. I will then rebase on it and see if we can get a clearer picture here.

Thanks!

**Comment by blueyed** [comment]
Rebased this.

**Comment by blueyed** [comment]
Too bad, https://codecov.io/gh/pytest-dev/pytest/pull/4829?src=pr&el=h1 is timing out, and codecov reports no status.

**Comment by blueyed** [comment]
But https://codecov.io/gh/pytest-dev/pytest/compare/4668ee0...0e7c3f7dd9904ff745d7a0ee9fd31278bf0f6864/diff works.

File: src/_pytest/python.py [hunk 3]
```diff
@@ -1419,16 +1414,6 @@ def __init__(
...
-        if self._isyieldedfunction():
-            assert not hasattr(
-                self, "callspec"
-            ), "yielded functions (deprecated) cannot have funcargs"
-        else:
-            if hasattr(self, "callspec"):
-                callspec = self.callspec
-                assert not callspec.funcargs
-                if hasattr(callspec, "param"):
-                    self.param = callspec.param
```
Comment (nicoddemus) [inline src/_pytest/python.py]: This no longer seems to be used anywhere, the `param` attribute would be added by `metafunc.addcall`, which no longer exists:

https://github.com/pytest-dev/pytest/blob/91b6f2bda8668e0f74190ed223a9eec01d09a0a7/_pytest/python.py#L888-L891

**Comment by nicoddemus** [comment]
@blueyed I have taken a look at `codecov/patch`, and they are all in `debugging.py`, which is unrelated to the code that I have removed. I understand the reasons for that now (thanks again for the patience and explanations), but I don't feel we should block PRs when the uncovered code by a patch are unrelated to the changes; I do think we should always try to improve overall coverage in dedicated PRs, I just don't feel we should incur that responsibility when the contributing code is not the cause for a decline in coverage.

What do you think?

**Comment by blueyed** [comment]
I think a red X should not mean that we cannot merge it, and it is a good indicator.
It is about code being touched, and if we/you think that it was uncovered before it is good to go in still.

**Comment by blueyed** [comment]
What is more strange are "changes", but that might be also due to codecov acting up..
"anypython" is considered to be covered now?!
https://codecov.io/gh/pytest-dev/pytest/compare/4668ee03f6bfb26dc4f8b89ddce3fdf54ea06288...0dcd808b45a6d9165eafb87629143b28bbc457c1/changes

A rebase might help here.

**Comment by nicoddemus** [comment]
OK, let's see how it goes then, rebased. 👍

**Comment by nicoddemus** [comment]
Travis failing because of https://github.com/pytest-dev/pytest/pull/4915... let's wait a little, but we can't wait too much as this will block builds from being merged. 😞

**Comment by blueyed** [comment]
> let's wait a little, but we can't wait too much as this will block builds from being merged.

Let's remove it from Travis already then tomorrow?
We can debug/fix it in the already open PR for it then.

**Comment by blueyed** [comment]
..but I think they might fix it soon already, it was broken only a few hours ago, inbetween some PRs/builds I did for pdbpp.

**Comment by nicoddemus** [comment]
> Let's remove it from Travis already then tomorrow?

Yeah I was thinking along the same lines. 😉 

> but I think they might fix it soon already

That would be great. I'm going to bed in an hour or so, so no rush from my part.

**Comment by nicoddemus** [comment]
Triggered pypy jobs manually on travis now that #4915 is working.

**Comment by nicoddemus** [comment]
Hmm coverage did not change in the end, even after the rebase. Anyway, merging as we discussed before. 👍

**Pull request merged**